app/jobs/scheduled/poll_mailbox.rb

```diff
@@ -96,12 +100,22 @@ def poll_pop3
           process_popmail(p)
         end
       end
-    rescue Net::OpenTimeout => e
-      mark_as_errored!
-      Discourse.handle_job_exception(e, error_context(@args, "Connecting to '#{SiteSetting.pop3_polling_host}' for polling emails."))
-    rescue Net::POPAuthenticationError => e
+    rescue Net::OpenTimeout, Net::POPAuthenticationError => e
+      if e.instance_of?(Net::OpenTimeout)
+        retry if tries < 3
+
+        notify_pop3_timeout_error
+        Discourse.handle_job_exception(e, error_context(@args, "Connecting to '#{SiteSetting.pop3_polling_host}' for polling emails."))
+      elsif e.instance_of?(Net::POPAuthenticationError)
+        if !$redis.get(POLL_MAILBOX_AUTHENTICATION_ERROR_KEY)
```

Comment:
@SamSaffron  I think email on authentication error should be sent immediately since it is unlikely that authentication errors are false positive.